### ISSUE TITLE
AttributeError: module has no attribute '__version__' only when running in ci (github actions)

### ISSUE THREAD

**Issue by CyrilWaechter** [pypa/cibuildwheel#775]
Hello,

I am trying to setup correctly a wheel build workflow for [python-fcl](https://github.com/CyrilWaechter/python-fcl) starting with linux only at the moment using github actions but I get errors I do not get locally. Help would be greatly appreciated.
# Structure explanation
Structure is very simple
`fcl.__init__.py` contains:
```python
from .version import __version__
```
`fcl.version` contains:
```python
__version__ = '0.6.1'
```
`setup.cfg` reference to version as following:
```
version = attr: fcl.__version__
```
Full workflow file can be found here: [wheels.yml](https://github.com/CyrilWaechter/python-fcl/blob/bf1d7cb3de795e09f4bb314f3e7866cebdfea1b4/.github/workflows/wheels.yml).

# Expected behaviour
I tried to locally reproduced the error but I was not able to. I tried:
- `python -m pip wheel ./ --wheel-dir=/tmp/cibuildwheel/built_wheel --no-deps`
- ` python -m build ./ --wheel --outdir=/tmp/cibuildwheel/built_wheel --config-setting=`

But both works so I expected same result during github actions build.

# Error message
```pytb
Building wheel...
  
      + rm -rf /tmp/cibuildwheel/built_wheel
      + mkdir -p /tmp/cibuildwheel/built_wheel
      + python -m build /project --wheel --outdir=/tmp/cibuildwheel/built_wheel --config-setting=
  Traceback (most recent call last):
    File "/opt/python/cp36-cp36m/lib/python3.6/site-packages/pep517/in_process/_in_process.py", line 280, in <module>
      main()
    File "/opt/python/cp36-cp36m/lib/python3.6/site-packages/pep517/in_process/_in_process.py", line 263, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/opt/python/cp36-cp36m/lib/python3.6/site-packages/pep517/in_process/_in_process.py", line 114, in get_requires_for_build_wheel
      return hook(config_settings)
    File "/tmp/build-env-2e6gxjxb/lib/python3.6/site-packages/setuptools/build_meta.py", line 155, in get_requires_for_build_wheel
      config_settings, requirements=['wheel'])
    File "/tmp/build-env-2e6gxjxb/lib/python3.6/site-packages/setuptools/build_meta.py", line 135, in _get_build_requires
      self.run_setup()
    File "/tmp/build-env-2e6gxjxb/lib/python3.6/site-packages/setuptools/build_meta.py", line 150, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 54, in <module>
      extra_compile_args = ["-std=c++11"]
    File "/tmp/build-env-2e6gxjxb/lib/python3.6/site-packages/setuptools/__init__.py", line 153, in setup
      return distutils.core.setup(**attrs)
    File "/opt/python/cp36-cp36m/lib/python3.6/distutils/core.py", line 121, in setup
      dist.parse_config_files()
    File "/tmp/build-env-2e6gxjxb/lib/python3.6/site-packages/setuptools/dist.py", line 779, in parse_config_files
      ignore_option_errors=ignore_option_errors)
    File "/tmp/build-env-2e6gxjxb/lib/python3.6/site-packages/setuptools/config.py", line 157, in parse_configuration
      meta.parse()
    File "/tmp/build-env-2e6gxjxb/lib/python3.6/site-packages/setuptools/config.py", line 463, in parse
      section_parser_method(section_options)
    File "/tmp/build-env-2e6gxjxb/lib/python3.6/site-packages/setuptools/config.py", line 436, in parse_section
      self[name] = value
    File "/tmp/build-env-2e6gxjxb/lib/python3.6/site-packages/setuptools/config.py", line 220, in __setitem__
      value = parser(value)
    File "/tmp/build-env-2e6gxjxb/lib/python3.6/site-packages/setuptools/config.py", line 557, in _parse_version
      version = self._parse_attr(value, self.package_dir)
    File "/tmp/build-env-2e6gxjxb/lib/python3.6/site-packages/setuptools/config.py", line 392, in _parse_attr
      return getattr(module, attr_name)
  AttributeError: module 'fcl' has no attribute '__version__'
  Collecting pip>=19.1.0
    Downloading https://files.pythonhosted.org/packages/47/ca/f0d790b6e18b3a6f3bd5e80c2ee4edbb5807286c21cdd0862ca933f751dd/pip-21.1.3-py3-none-any.whl (1.5MB)
  Installing collected packages: pip
    Found existing installation: pip 18.1
      Uninstalling pip-18.1:
        Successfully uninstalled pip-18.1
  Successfully installed pip-21.1.3
  Found existing installation: setuptools 40.6.2
  Uninstalling setuptools-40.6.2:
    Successfully uninstalled setuptools-40.6.2
  Collecting wheel
    Downloading wheel-0.36.2-py2.py3-none-any.whl (35 kB)
  Collecting Cython
    Using cached Cython-0.29.24-cp36-cp36m-manylinux1_x86_64.whl (2.0 MB)
  Collecting setuptools
    Downloading setuptools-57.4.0-py3-none-any.whl (819 kB)
  Collecting numpy
    Using cached numpy-1.19.5-cp36-cp36m-manylinux2010_x86_64.whl (14.8 MB)
  Installing collected packages: wheel, setuptools, numpy, Cython
  Successfully installed Cython-0.29.24 numpy-1.19.5 setuptools-57.4.0 wheel-0.36.2
  
  ERROR Backend subproccess exited when trying to invoke get_requires_for_build_wheel
                                                             ✕ 12.65s
Checking for common errors...
Error: Command ['python', '-m', 'build', PurePosixPath('/project'), '--wheel', '--outdir=/tmp/cibuildwheel/built_wheel', '--config-setting='] failed with code 1. 

Error: Process completed with exit code 1.
```

**Comment by henryiii** [comment]
First think I'd do is switch to using the configuration mode (in pyproject.toml) instead of environment variables to configure cibuildwheel. Then you can try it locally with `cibuildwheel --platform linux` and that's likely easier to debug.

**Comment by henryiii** [comment]
Also, you should remove the setup-python line in GitHub actions, it's not doing anything.

First thing I'd try is to change `version = attr: fcl.__version__` to `version = attr: fcl.version.__version__`, as newer versions of setuptools will try to parse the AST to get the version, so the file it points at should be as simple as possible; it can't follow an import.

**Comment by CyrilWaechter** [comment]
I moved `cibuildwheel` configuration to `pyproject.toml` as advised. I am now able to reproduce the error locally using `cibuildwheel ./ --platform=linux`.
I changed `version = attr: fcl.__version__` to `version = attr: fcl.version.__version__` but I get a similar error:
`ModuleNotFoundError: No module named 'fcl.version`.

**Comment by joerick** [comment]
FYI @CyrilWaechter I was able to recreate this error above on my local machine. I downloaded a zip file snapshot of 'bf1d7cb3de795e09f4bb314f3e7866cebdfea1b4', then made a new Python 3.8 virtualenv, and did 

```console
bash requirements/clone.bash
bash requirements/build.bash
python -m pip wheel ./ --wheel-dir=/tmp/cibuildwheel/built_wheel --no-deps
```

I was given this error message:
```
(tempenv) joerick@joerick4 ~/D/python-fcl-bf1d7cb3de795e09f4bb314f3e7866cebdfea1b4> python -m pip wheel ./ --wheel-dir=/tmp/cibuildwheel/built_wheel --no-deps
Processing /Users/joerick/Downloads/python-fcl-bf1d7cb3de795e09f4bb314f3e7866cebdfea1b4
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: /private/var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/tmp.7uy9F9k0/tempenv/bin/python /private/var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/tmp.7uy9F9k0/tempenv/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py get_requires_for_build_wheel /var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/tmpe72i1b9l
       cwd: /private/var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/pip-req-build-7rcmvi19
  Complete output (34 lines):
  Traceback (most recent call last):
    File "/private/var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/tmp.7uy9F9k0/tempenv/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 280, in <module>
      main()
    File "/private/var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/tmp.7uy9F9k0/tempenv/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 263, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/private/var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/tmp.7uy9F9k0/tempenv/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 114, in get_requires_for_build_wheel
      return hook(config_settings)
    File "/private/var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/pip-build-env-e1g7l2f7/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 154, in get_requires_for_build_wheel
      return self._get_build_requires(
    File "/private/var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/pip-build-env-e1g7l2f7/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 135, in _get_build_requires
      self.run_setup()
    File "/private/var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/pip-build-env-e1g7l2f7/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 150, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 44, in <module>
      setup(
    File "/private/var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/pip-build-env-e1g7l2f7/overlay/lib/python3.8/site-packages/setuptools/__init__.py", line 153, in setup
      return distutils.core.setup(**attrs)
    File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/distutils/core.py", line 121, in setup
      dist.parse_config_files()
    File "/private/var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/pip-build-env-e1g7l2f7/overlay/lib/python3.8/site-packages/setuptools/dist.py", line 778, in parse_config_files
      parse_configuration(self, self.command_options,
    File "/private/var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/pip-build-env-e1g7l2f7/overlay/lib/python3.8/site-packages/setuptools/config.py", line 157, in parse_configuration
      meta.parse()
    File "/private/var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/pip-build-env-e1g7l2f7/overlay/lib/python3.8/site-packages/setuptools/config.py", line 463, in parse
      section_parser_method(section_options)
    File "/private/var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/pip-build-env-e1g7l2f7/overlay/lib/python3.8/site-packages/setuptools/config.py", line 436, in parse_section
      self[name] = value
    File "/private/var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/pip-build-env-e1g7l2f7/overlay/lib/python3.8/site-packages/setuptools/config.py", line 220, in __setitem__
      value = parser(value)
    File "/private/var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/pip-build-env-e1g7l2f7/overlay/lib/python3.8/site-packages/setuptools/config.py", line 557, in _parse_version
      version = self._parse_attr(value, self.package_dir)
    File "/private/var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/pip-build-env-e1g7l2f7/overlay/lib/python3.8/site-packages/setuptools/config.py", line 392, in _parse_attr
      return getattr(module, attr_name)
  AttributeError: module 'fcl' has no attribute '__version__'
  ----------------------------------------
ERROR: Command errored out with exit status 1: /private/var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/tmp.7uy9F9k0/tempenv/bin/python /private/var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/tmp.7uy9F9k0/tempenv/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py get_requires_for_build_wheel /var/folders/ld/k24nt7054698bctspqwrjq1r0000gn/T/tmpe72i1b9l Check the logs for full command output.
```

Maybe you have the fcl package installed in your virtualenv, so it can find it? On my machine, the fcl dir doesn't contain a python package, it's like this:

```
(tempenv) joerick@joerick4 ~/D/python-fcl-bf1d7cb3de795e09f4bb314f3e7866cebdfea1b4> ls fcl
CHANGELOG.md             INSTALL                  fcl-targets.cmake
CMakeCache.txt           LICENSE                  fcl.pc
CMakeFiles               Makefile                 fcl.pc.in
CMakeLists.txt           README.md                include
CMakeModules             Testing                  install_manifest.txt
CPPLINT.cfg              ci                       lib
CTestConfig.cmake        cmake_install.cmake      package.xml
CTestCustom.cmake        doc                      src
CTestCustom.cmake.in     fcl-config-version.cmake test
CTestTestfile.cmake      fcl-config.cmake         test_results
DartConfiguration.tcl    fcl-config.cmake.in
```

**Comment by henryiii** [comment]
`clone.bash` contains:

```

rm -rf fcl
git clone --depth 1 --branch v0.6.1 https://github.com/flexible-collision-library/fcl.git
```

So that's why, you are deleting the whole folder then replacing it.

**Comment by CyrilWaechter** [comment]
I totally missed the naming conflict between python module name and its c++ dependency. Thanks a lot @henryiii and @joerick.

Now I have another issue during test phase. Missing symbol…

**Comment by Czaki** [comment]
Are you sure that using the same names as in fcl c++ code is a good idea? Maybe it is some name collision? 

Try to install the same gcc version locally and try to reproduce this error.

**Comment by CyrilWaechter** [comment]
> Are you sure that using the same names as in fcl c++ code is a good idea? Maybe it is some name collision?

@Czaki I fixed the name collision by building dependencies in their own folder. I am not the author of python-fcl. I just fixed it to work with new version of fcl (0.6.1) but I do not see it as an issue as python-fcl is python library with no executable. Why would it be wrong ? Isn’t it common for a wrapper/binding library ?

> Try to install the same gcc version locally and try to reproduce this error.

Ok. I’ll try this.

**Comment by CyrilWaechter** [comment]
Using `manylinux_2010` instead of `manylinux_2_24` fixed the issue.
